cpp/src/utilities/wrapper_types.hpp

```diff
@@ -263,6 +263,9 @@ using date32 = detail::wrapper<gdf_date32, GDF_DATE32>;
 
 using date64 = detail::wrapper<gdf_date64, GDF_DATE64>;
 
+using gdf_bool 	= detail::wrapper<int8_t, GDF_BOOL>;
```

Comment:
I believe it would be problematic to use `bool` natively,  because the size of `bool` in C++ is implementation defined.

Reply:
Of course it would! ... I was just being consistent with the other problematic typedef's:
```
typedef int gdf_size_type; /**< Limits the maximum size of a gdf_column to 2^31-1 */
typedef gdf_size_type gdf_index_type;
typedef unsigned char gdf_valid_type;
typedef	long	gdf_date64;
typedef	int		gdf_date32;
typedef	int		gdf_category;
typedef	long	gdf_timestamp;
```
`int` and `long` are also of implementation-defined size... I'd be all for 

typedef int32_t gdf_size_type;
typedef gdf_size_type gdf_index_type;
typedef uint8_t gdf_valid_type; /* shouldn't this become uint32_t ? */
typedef int64_t gdf_date64;
typedef int32_t gdf_date32;
typedef int64_t gdf_timestamp;
typedef int32_t gdf_category;
typedef int8_t gdf_bool;

Reply:
The initial conversation was around rather to simply do a `typedef int8_t gdf_bool` or to use the    type_dispatcher.  We had decided to go down the type_dispatcher route, hence the is no typedef.

Reply:
@BradReesWork : I don't quite understand. How does it make sense to have `typedef int gdf_date32;` but not have `typedef signed char gdf_bool` or `typedef int8_t gdf_bool`?